[Agent][Python 3] Add Agent configuration documentation for Python 3 (#5334)
DataDog/documentation: content/en/agent/guide/python-3.md

To provide flexibility in allowing code to run multiple on versions of the Agent, this guide focuses on retaining backwards compatibility.

## Agent configuration

> Starting with v6.14.0, the Agent integrates both Python 2 and Python 3 runtimes. This means that custom Agent Checks can be run either with Python 2 or Python 3, depending on the Agent configuration.

Reviewer: I'd remove `custom` here, since it applies to both custom and official checks
```suggestion
Starting with v6.14.0, the Agent integrates both Python 2 and Python 3 runtimes. This means that Agent Checks can be run either with Python 2 or Python 3, depending on the Agent configuration.
```

By default, the Agent v6 uses the Python 2 runtime. To switch to the Python 3 runtime:

1. Set the `python_version` configuration option [in your `datadog.yaml` configuration file][1]:

```yaml
python_version: 3
```

2. [Restart the Agent][2].

> Alternatively, the `DD_PYTHON_VERSION` environment variable can be set to `2` or `3` to choose which Python runtime is used. If it is set, the `python_version` option in `datadog.yaml` will be ignored.

Reviewer 2: ```suggestion
Alternatively, the `DD_PYTHON_VERSION` environment variable can be set to `2` or `3` to choose which Python runtime is used. If it is set, the `python_version` option in `datadog.yaml` is ignored.
```

This is an Agent-wide configuration option: **all Python checks launched by an Agent use the same Python runtime**.

### Containerized Agent

The official Containerized Agent images only include one of the two Python runtimes. To switch to one of the other runtime, choose the appropriate Agent image:

> * **Python 2** runtime: Agent v6 images have the following format: `datadog/agent:<VERSION>`, or `datadog/agent:<VERSION>-jmx` for images supporting JMX checks.

Reviewer 2: ```suggestion
* **Python 2** runtime: Agent v6 images have the following format: `datadog/agent:<AGENT_VERSION>`, or `datadog/agent:<AGENT_VERSION>-jmx` for images supporting JMX checks.
```

> * **Python 3** runtime: Agent v6 images have the following format: `datadog/agent:<VERSION>-py3`, or `datadog/agent:<VERSION>-py3-jmx` for images supporting JMX checks.

Reviewer 2: ```suggestion
* **Python 3** runtime: Agent v6 images have the following format: `datadog/agent:<AGENT_VERSION>-py3`, or `datadog/agent:<AGENT_VERSION>-py3-jmx` for images supporting JMX checks.
```

To provide flexibility in allowing code to run multiple on versions of the Agent, this guide focuses on retaining backwards compatibility.

## Agent configuration

> Starting with v6.14.0, the Agent integrates both Python 2 and Python 3 runtimes. This means that checks can be run with either Python 2 or Python 3, depending on the Agent configuration.

Reviewer 2: ```suggestion
Starting with v6.14.0, the Agent integrates both Python 2 and Python 3 runtimes. This means that custom Agent Checks can be run either with Python 2 or Python 3, depending on the Agent configuration.
```

> By default, the Agent v6 will use the Python 2 runtime. To switch to the Python 3 runtime, set the `python_version` configuration option in the `datadog.yaml` configuration file:

Reviewer 2: ```suggestion
By default, the Agent v6 uses the Python 2 runtime. To switch to the Python 3 runtime, set the `python_version` configuration option [in the `datadog.yaml` configuration file](/agent/guide/agent-configuration-files/?tab=agentv6#agent-main-configuration-file):
```

> By default, the Agent v6 will use the Python 2 runtime. To switch to the Python 3 runtime, set the `python_version` configuration option in the `datadog.yaml` configuration file:

Reviewer 2: I think we should do a numbered bullet point list, like:

1. Set the ...
2. [Restart the Agent](/agent/guide/agent-commands/?tab=agentv6#restart-the-agent).

```yaml
python_version: 3
```

and restart the Agent.

> Alternatively, the `DD_PYTHON_VERSION` environment variable can be set to `2` or `3` to choose which Python runtime is used. If it is set, the `python_version` option in `datadog.yaml` will be ignored.

Reviewer 2: This new env variable should be documented here: https://docs.datadoghq.com/agent/docker/?tab=standard#environment-variables as well.

Author: I don't think this option should be documented in the Docker Agent's pages, since it should *not* be used with the Containerized Agent (as mentioned in the next section, you choose your python version by choosing a different Docker Agent image depending on whether you use Python 2 or Python 3 ; using the `DD_PYTHON_VERSION` option with the Docker Agent could actually be harmful, since you could set the option to a Python version that is not in the image, which would prevent any Python check from running).

I do think however that we could add a link to the Containerized Agent section of the Python 3 page in the Docker Agent's page, so that people reading the Docker Agent doc can easily know how to have an image running Python 3.

> This is an Agent-wide configuration option: all Python checks launched by an Agent will use the same Python runtime.

Reviewer 2: ```suggestion
This is an Agent-wide configuration option: **all Python checks launched by an Agent use the same Python runtime**.
```
